worlds/tunic/er_scripts.py

```diff
@@ -466,6 +466,54 @@ def add_dependent_regions(region_name: str, dependent_regions: Dict[Tuple[str, .
     return region_set
 
 
+def update_reachable_regions(connected_regions: Set[str], traversal_reqs: Dict[str, Dict[str, List[List[str]]]], has_laurels: bool) -> Set[str]:
+    # starting count, so we can run it again if this changes
+    region_count = len(connected_regions)
+    # checking each origin region
+    for origin, destinations in traversal_reqs.items():
+        # if the origin is not connected, skip it
+        if origin not in connected_regions:
+            continue
+        # check if we can traverse to any of the destinations
+        for destination, req_lists in destinations.items():
+            # if the destination already connected, skip it
+            if destination in connected_regions:
+                continue
+            # met meaning met the requirements to traverse
+            met = False
+            # if the list of requirement lists is empty, then you can freely connect this
+            if len(req_lists) == 0:
+                met = True
+            # loop through each set of possible requirements
+            for reqs in req_lists:
+                # if for some reason a list is empty, then you can freely connect it
+                if len(reqs) == 0:
+                    met = True
+                else:
+                    met_count = 0
+                    for req in reqs:
+                        # will need to change this if item plando gets evaluated earlier
+                        if req == "Hyperdash" and has_laurels:
+                            met_count += 1
+                        # check if we have the regions required for traversal
+                        if req in connected_regions:
+                            met_count += 1
+                    if met_count == len(reqs):
+                        met = True
+                # if we met one set of requirements, no need to check the next set
+                if met == True:
+                    break
```

Comment:
How do you mean?

Reply:
```
any(
    all(
        req == "Hyperdash" and has_laurels or req in connected_regions
        for req in reqs
    )
    for reqs in reqs_list
)
```

Reply:
I think I'm gonna have to decline for now at least -- maybe later on when I'm a bit more comfortable with stuff like this

Reply:
I'm not gonna force the any(all()) thing, but even with for loops, I'm not happy with this particular implementation

I would much prefer not explicitly checking for the 0 case because it already evaluates correctly without it

Ontop of that, the "met count" thing is not great either imo.

What you're currently asking is "are as many parts of this valid as there are parts". The much simpler thing (imo) to ask is "is any part of this **invalid**"
Here's what I would do.

```
met_traversal_reqs = False
for reqs in req_lists:
    current_reqs_is_valid = True
    for req in reqs:
        # will need to change this if item plando gets evaluated earlier
        if req == "Hyperdash" and not has_laurels:
            current_reqs_is_valid = False
            break
        if req == "NMG" and not logic:
            current_reqs_is_valid = False
            break
        if req == "UR" and logic != 2:
            current_reqs_is_valid = False
            break
        if req not in connected_regions:
            current_reqs_is_valid = False
            break
    if reqs_is_valid:
        met_traversal_reqs = True
        break
```

or, if you're willing to go a bit dank, this would be the shortest for loop version that keeps the requirements individual

```
met_traversal_reqs = False
for reqs in req_lists:
    for req in reqs:
        # will need to change this if item plando gets evaluated earlier
        if req == "Hyperdash" and not has_laurels:
            break
        if req == "NMG" and not logic:
            break
        if req == "UR" and logic != 2:
            break
        if req not in connected_regions:
            break
    else:  # Evaluates only if for loop was not broken out of
        met_traversal_reqs = True
        break
```

Reply:
I'm going to be honest, it felt a little strange clicking "unresolve" on this. I am 100% sympathetic to "I'm not comfortable using your suggestion because it's not intuitive to me", but I wish you would've let Ixrec make a counter-suggestion

Unless Ixrec is the one who clicked "Resolve", in which case, disregard this

Reply:
I don't want to make a huge deal out of it though, it was one button click. It just gave me pause and I wanted to explain my feelings.

Reply:
FWIW, I did not click resolve, or do anything else to the status of this comment chain. Since there's no risk of harm beyond code readability, I considered my reviewer duty complete once it was clear Scipio understood the suggestion and had made the informed choice to not adopt it. My previous approval was meant to imply I was satisfied with the state of all review comment chains.

Reply:
I only hit resolve after a couple days, not immediately. And I believe the original reviewer can un-resolve it. I see hitting resolve as "I have made a decision regarding this review comment, and now want to collapse it so that it doesn't look like it's an open item that I need to act upon". Which I wish I could do with the suggestion at the bottom here since I've already acted upon it, since it's making me anxious looking at it even though it's done